crates/gitbutler-core/src/git/diff.rs

```diff
@@ -101,6 +101,22 @@ impl GitHunk {
     }
 }
 
+/// Comparison
+impl GitHunk {
+    /// integration_intersects_unapplied is used to determine if a hunk from a diff between integration and the trunk intersects with an unapplied hunk.
+    /// We want to use the new start/end for the integraiton hunk and the old start/end for the unapplied hunk.
+    pub fn integration_intersects_unapplied(
+        integration_hunk: &GitHunk,
```

Comment:
I think when the first argument is also a `GitHunk`, it should really be `&self` to allow calling it more naturally.
With an adjusted method name it should also 'sound' natural.

Reply:
Hi @Byron, this was indeed deliberate because I wanted to clearly name the roles of each hunk you're passing in because its very important in this case.

I could define some `struct IntegrationHunk(GitHunk)` which has the method on it but it felt a bit excessive